rest_framework/schemas/openapi.py

```diff
@@ -101,6 +112,34 @@ def get_operation(self, path, method):
 
         return operation
 
+    def _get_serializer_component_name(self, serializer):
+        if not hasattr(serializer, 'Meta'):
+            return None
+
+        if hasattr(serializer.Meta, 'schema_component_name'):
+            return serializer.Meta.schema_component_name
```

Comment:
Drop this. Instead have AutoSchema take an `__init__()` kwarg, `component_name`. If that's passed use it. Otherwise generate a name as per below. 

Usage would be: 

```
class MyView(APIView):
   schema = AutoSchema(component_name="Ulysses")
```

If folks need more complex usage, they can override the method.